File: workspaces-good/packages/shared/src/taxonomy_expression_vo.ts
import { OperationVO } from "./taxonomy_operation_vo";

export interface ExpressionVO {
  left: number;
  op: OperationVO;
  right: number;
}

export function createExpression(
  left: number,
  op: OperationVO,
  right: number,
): ExpressionVO {
  return { left, op, right };
}

File: workspaces-good/modules/shared/src/taxonomy_expression_vo.py
from .taxonomy_operation_vo import OperationVO


class ExpressionVO:
    def __init__(self, left: float, op: OperationVO, right: float):
        self.left = left
        self.op = op
        self.right = right

    def __repr__(self) -> str:
        return f"{self.left} {self.op.value} {self.right}"


def create_expression(left: float, op: OperationVO, right: float) -> ExpressionVO:
    return ExpressionVO(left, op, right)
